### ISSUE TITLE
2x requests - Avoid importing duplicated accounts & export nonencrypted backup

### ISSUE THREAD

**Issue by anogues** [nuxsmin/sysPass#376]
Hello

I have two low priority requests

-  When importing a csv file, if an account is duplicated, it creates two accounts, with possibily a different password. I think it should either throw a warning or update the previously found account. We thought about putting a unique key in the accounts table by several of the columns, but there should probably be a more elegant way to do it :)

- * Export passwords in clear text for a backup, just in case the master password gets lost or something similar. Preferently in csv, but current xml file will do as well.

**Comment by anogues** [comment]
Another two hehe

- Fix the main ui so the headers are aligned with the content. See the picture
![image](https://cloud.githubusercontent.com/assets/8581047/22147714/42ddcb2e-df0a-11e6-8e9a-0019a7ec1818.png)

- Add notes section or a part of them in the main window. So in the search screen, where all the accounts are listed. I understand this can be a bit of a mess if the notes field is large, so not sure if possible to do it.

**Comment by LeDistordu** [comment]
+1 Fix the main ui so the headers are aligned with the content

**Comment by nuxsmin** [comment]
Hello, that UI behavior is a misunderstood behavior, because, in a future, some additional sorting fields could be added, but they may not be paired with a field in the account row, so the alignment wouldn't make sense. Those additional headers could be like: number of views, number of files, etc...

**Comment by anogues** [comment]
Ok, I see the UI thing. That is OK.

Most urgent one for us is the duplicated account thing. In fact i don't think it makes too much sense to have the same account, icnluding user with a different password (well it may have, if you have local and domain accounts for example...) but i think it will be good to prevent accidental duplicates

**Comment by nuxsmin** [comment]
Fine, but you will lost those duplicated accounts that came from the CSV, and then...why are they there?.

i.e: I've many duplicated accounts name because linux servers (through ftp, ssh, etc) have many users to the same server, so the account name/customer/url (and even category) are the same for those entries.

**Comment by anogues** [comment]
Well, i see your point. But at this point we are doing a bulk load of several password files we have, and we know there are duplicated accounts in these. One option is to clean manually duplicates in the file which is feasible but can take a considerable ammount of time.

We are still thinking about it. For later it is not a problem, because we have added a unique key in the account table, and if the user tries to add a duplicate account the database sees its and returns an error which is more than fine.

Obviously with the import process it does not work because it fails at some point and we dont know until which point of the file the import process worked, but for adding them manualy it works fine.

So we solved half of the problem. But again, this is not very urgent. I understand there are topics in the list that may be more interesting / have wider use cases.

Regards!
Albert.

**Comment by akzov** [comment]
If you import passwords you'll never be able to use a different password manager unless you type each password manually.

**Comment by nuxsmin** [comment]
@akzov please be patient, with some tweaks the tool https://github.com/julenpardo/sysPass-XML-Decrypter will be integrated into sysPass to get an easy way to export your accounts.

By the way, giving an easy way to export the data does not mean that you will be able to import it in another tool, because sysPass uses its own data structure, so you will need to adapt to the target application. Some examples: what will you do with tags? and what about custom fields?

To sum up, there won't be an universal export-import tool.
